libraries/Microsoft.Bot.Builder/ShowTypingMiddleware.cs

```diff
@@ -165,8 +165,11 @@ private async Task FinishTypingTaskAsync(ITurnContext turnContext)
             cts?.Dispose();
             if (typingTask != null)
             {
-                await typingTask.ConfigureAwait(false);
-                typingTask.Dispose();
+                if (!typingTask.IsFaulted)
+                {
+                    await typingTask.ConfigureAwait(false);
+                    typingTask.Dispose();
+                }
```

Comment:
Why nesting the if when there's no other possible path of execution? Couldn't we just use && ?

Reply:
you're right, fixed